vendor-bin/box/composer.json

```diff
@@ -0,0 +1,6 @@
+{
+    "require": {
+        "humbug/box": "^3.0@dev",
+        "humbug/php-scoper": "^1.0@dev"
```

Comment:
if we are not scoping, it's not needed, right?

Reply:
It's a Box dependency, unfortunately as PHP-Scoper is not in a stable version so just requiring Box will fail saying no stable dependency could be found for PHP-Scoper.

Alternatively I can just add `minimal-stability: dev` with `prefer-stable: true` to remove it from the `composer.json`

Reply:
if using single unstable release is not convincing me, then allowing for using **any** won't convince me either ;)

Reply:
maybe `php-scoper` shall be optional for `box` in that case ?

Reply:
both dependencies are unstable and there, it's just a matter of how you want to display & control them (and in your case you don't care about PHP-Scoper).

Be it PHP-Scoper or some other dependencies which are actually not needed until you use a specific feature of Box, I don't see any easy way to do it.

Reply:
super simple example how we did it with optional extension (but it can be done the same with optional library): https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.11/src/Report/XmlReporter.php#L37

it's matters especially with extension, we don't want to force all users to have the extension, if just some of them will use it.

Reply:
but yeah, will look definitely better when it become stable with no need to do hacks like that.